### PR TITLE
spark-hdfs back to normal

### PR DESCRIPTION
**Pull request by rabbitonweb**
This is part of #2471 
It upgrades spark to latest `2.2.0` + brings spark-hdfs back to normal

### CODE REVIEW

File: frontend/src/main/scala/quasar/std/date.scala [hunk 1]
```diff
@@ -168,55 +168,55 @@ trait DateLib extends Library with Serializable {
       constTyper[nat._1](Type.Numeric),
       basicUntyper)
 
-  val ExtractCentury      = extract(
+  lazy val ExtractCentury      = extract(
```
Comment (rintcius) [inline frontend/src/main/scala/quasar/std/date.scala]: can you add a note (+ link to SO question) why this is needed?

**Comment by rintcius** [comment]
Oh, can you also add `version : ...` label?

File: frontend/src/main/scala/quasar/std/date.scala (hunk 1 above)
Comment (rabbitonweb) [inline frontend/src/main/scala/quasar/std/date.scala]: It is still not completely clear for me why this is needed. In essence we had error manifasting itself as `ClassNotFoundError` on spark's executors. This normally means that sparkcore.jar (uberjar) is not send to the cluster or the given class (`DateLib`) is not in that jar. After verifying that is not the case, I kept on googling and I found this https://stackoverflow.com/questions/7262799/why-does-scala-fail-to-instantiate-a-companion-object Seemed odd but it was worth a try. And it solved the issue. This is something worth investigating further and I'm actually planning to do so (recreate the issue on a sandboxed project), but for now I need working spark connectors and this thing solves it (even though it runs on magic at the moment)

Comment (rabbitonweb) [inline frontend/src/main/scala/quasar/std/date.scala]: Please note that
1. Any sufficiently advanced technology is indistinguishable from magic
2. spark does a lot of dynamic loading using classloaders and we might have fallen to one of the (not yet discovered) bugs

**Pull request merged**